BudgetTracker.BudgetSquirrel.WebApi/Controllers/AuthenticationApiController.cs

```diff
@@ -64,55 +56,46 @@ public async Task<IActionResult> DeleteUser(ApiRequest request)
         }
 
         /// <summary>
-        /// Returns a new token with claims information for the user specified
-        /// in the request. This token should be attached as a bearer token in
-        /// sub-sequent requests that require authentication.
+        /// Returns a new cookie with claims information for the user specified
+        /// in the request. This cookie will persistent until the user logs out
+        /// or the session times out.
         /// </summary>
-        /// <param name="data">
-        /// A JSON object that contains 2 keys:
-        /// - username
-        /// - password
-        /// These are used to authenticate the user for which this token is generated.
-        /// </param>
+        /// <param name="credentials">The username password used for the attempt to login <see cref="Credentials"/></param>
         /// <returns>
-        /// A JWT Token with claims information for the authenticated user. If
+        /// A Cookie with claims information for the authenticated user. If
         /// the user cannot be authenticated, a 403 will be thrown.
         /// </returns>
         [HttpPost("authenticate")]
-        public async Task<IActionResult> Authenticate(Dictionary<string, string> data)
+        public async Task<IActionResult> Authenticate(Credentials credentials)
         {
             User authenticatedUser;
             UserResponseMessage userResponse;
             try
             {
-                authenticatedUser = await _authenticationService.Authenticate(data["username"], data["password"]);
+                authenticatedUser = await _authenticationService.Authenticate(credentials.Username, credentials.Password);
                 userResponse = new UserResponseMessage(authenticatedUser);
             }
             catch (AuthenticationException e)
             {
-                return new JsonResult(new {
+                return this.BadRequest(new JsonResult(new {
                     error = e.Message
```

Comment:
Same thing, shouldn't this just return a message that says "Invalid login"?

Reply:
e.Message contains the exact issue that occurred... So like it might say "user does not exist" which leads to a user farming issue